src/strategyHelpers/deploymentHelper.ts

```diff
@@ -59,16 +59,17 @@ export async function deployManifests(
             core.getInput('route-method', {required: true})
          )
 
-         const {result, newFilePaths} = await Promise.resolve(
+         const {result, newObjectsList} = await Promise.resolve(
             (routeStrategy == RouteStrategy.INGRESS &&
-               deployBlueGreenIngress(kubectl, files)) ||
+               (deployBlueGreenIngress(kubectl, files))) ||
                (routeStrategy == RouteStrategy.SMI &&
                   deployBlueGreenSMI(kubectl, files)) ||
                deployBlueGreenService(kubectl, files)
          )
 
-         checkForErrors([result])
-         return newFilePaths
+
+         checkForErrors([result.result])
```

Comment:
this pattern of result.result is rough. Better names for things will help a ton. You might want to flatten out this object